presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkDistribution.java

```diff
@@ -49,13 +51,38 @@ public PrestoSparkDistribution(
             Optional<Map<String, String>> sessionPropertyConfigurationProperties,
             Optional<Map<String, Map<String, String>>> functionNamespaceProperties,
             Optional<Map<String, Map<String, String>>> tempStorageProperties)
+    {
+        this(
+                sparkContext,
+                packageSupplier,
+                configProperties,
+                catalogProperties,
+                ImmutableMap.of(METADATA_STORAGE_TYPE_KEY, metadataStorageType),
+                eventListenerProperties,
+                accessControlProperties,
+                sessionPropertyConfigurationProperties,
+                functionNamespaceProperties,
+                tempStorageProperties);
+    }
```

Comment:
Should we remove just this constructor? This is an internal API which is being changed. So, we can just remove the code completely instead of marking it as deprecated?

Reply:
I don't think this is purely internal. It's a public constructor that might be used in many places already. I feel it's safer not to make breaking changes. What do you think @highker ?

Reply:
I get your point. This API can be called from external extensions (like the one we have within Meta) and that might be dependent on the old API. Let's keep it for now.